test/runtime/runtime-cplusplus/src/AddressTest.cpp

```diff
@@ -0,0 +1,84 @@
+
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include "utils/doctest.h"
+#include "utils/nlohmann/json.hpp"
+
+#include <string>
+#include <vector>
+
+#include "lib/generated/address.hpp"
+
+using json = nlohmann::json;
+
+TEST_CASE("Should be able to serialize C++ model")
+{
+    AsyncapiModels::address address;
+    address.street_name = "Test address 2";
+    address.house_number = 2.0;
+    address.marriage = true;
+    address.members = 2.0;
+    std::vector<std::variant<std::string, double, std::any>> array_type;
+    array_type.push_back(2.0);
+    array_type.push_back(std::string("test"));
+    address.array_type = array_type;
+    AsyncapiModels::nested_object obj;
+    obj.test = "test";
+    address.nested_object = obj;
+
+    std::vector<std::string> stringVector;
+    for (const auto &item : array_type)
+    {
+        if (auto str = std::get_if<std::string>(&item))
+        {
+            stringVector.push_back(*str);
+        }
+    }
+
+    json jsonData;
+    jsonData["streetName"] = address.street_name;
+    jsonData["houseNumber"] = address.house_number;
+    jsonData["marriage"] = address.marriage.value();
+    jsonData["members"] = std::get<double>(address.members.value());
+    jsonData["arrayType"] = stringVector;
+    jsonData["nestedObject"] = {{"test", address.nested_object.value().test.value()}};
+
+    std::string jsonStr = jsonData.dump();
```

Comment:
This entire code should actually be part of the generated models themself, cause right now it's your own implementation of the serialization that you are testing.

This does not test the generated output in the slightest 🙂

Reply:
Now that you've mentioned it. I kind of feel dumb when rereading the issue.

So I should just create a object of that class in single test?

Also if needed I can help with serialization of c++ modules to json as I've already looked this much into it with this issue

Reply:
@jonaslagoni any inputs?

Reply:
Sorry, must have missed this question 🙇 

> So I should just create a object of that class in single test?

Yes, make it easy first test. It might feel useless, however, see this PR as the starting point and setup for C++ runtime testing, and not so much about creating a whole set of tests 🙂 

If you want, you are more then welcome to do it afterwards, just keep each PR as small as possible 🙂